### PR TITLE
DOC: Correction for scipy.stats.poisson - interval documentation

### PR DESCRIPTION
**Pull request by harmbuisman**
Docs of poisson interval mention percentage, which suggests number between 0 and 100. Corrected docs to mention fraction and added interval [0, 1].

### CODE REVIEW

**Comment by chrisb83** [comment]
`interval` is a method of all distributions in SciPy. I don't understand the current and proposed description, should it be "Confidence interval with equal areas around the median.", see
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.html#scipy.stats.rv_continuous

**Comment by josef-pkt** [comment]
The short description in the specific class differs from the one in rv_continuous

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html#scipy.stats.norm

The current "percent" is incorrect if it is specified as probability like 0.95

I think "equal areas" is not really good because I don't know what "area" this refers to.
In the continuous case, it is a "equal tail interval that covers alpha probability", AFAIR.
in the discrete case the coverage should be at least alpha, but will in general be larger because of the discrete sample space, and possible corner cases (where one tail has 0 probability, i.e. is empty)

**Comment by pvanmulbregt** [comment]
There are two different docstring descriptions for `interval`.

There is a default `_doc_interval` specified in `scipy/stats/_distn_infrastructure.py:138`, which contains "Endpoints of the range that contains alpha percent of the distribution".
Separately `rv_generic.interval()` has a docstring which starts "Confidence interval with equal areas around the median."

In the generated docs, `rv_discrete` and `rv_continuous` both have the same description as rv_generic, namely "Confidence...".  And descriptions of `interval` for the individual distributions have the `_doc_interval` "Endpoints ...".
But inside a python session, the docstrings for *all* of the `XXX.interval()` methods (as shown via  `help(scipy.stats.norm.interval)`) is the "Confidence interval with equal areas around the median." string.

I.e. It appears that there are different descriptions displayed depending on whether one is inside a Python session or one is reading the generated docs. 
@chrisb83 Do you see this as well or is this just some artifact of my current build environment?

**Comment by pvanmulbregt** [comment]
By "generated docs" I meant the generated HTML  The description of norm.interval in https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html has the other docstring containing "percent".

**Comment by chrisb83** [comment]
I think the method `interval` is documented in `rv_generic` (~line 1264 in _distn_infrastructure.py) with the statement "Confidence interval with equal areas around the median.", i.e. this is what you see in the generated docs of `rv_continuous` and `rv_discrete` and in your python session if you type `help(norm.stats.interval)`.

`_doc_interval` is used in the generated doc of the distribution (i.e. not the method `interval`) that plots the pdf, summarizes the methods etc

**Comment by pvanmulbregt** [comment]
Which raises the question as to why we have two different docstrings describing the functionality? And is there a way to share the content between the two use cases?

**Comment by mdhaber** [comment]
To summarize, in `distn_infrastructure.py`, there is:
```
_doc_interval = """\
interval(alpha, %(shapes)s, loc=0, scale=1)
    Endpoints of the range that contains alpha percent of the distribution
"""
```
Which renders in each distribution's documentation (e.g. [Poisson](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.poisson.html#scipy.stats.poisson)) like:
image


Separately, there is the documentation of `rv_continuous`, which comes from its docstring:
```
    def interval(self, alpha, *args, **kwds):
        """
        Confidence interval with equal areas around the median.
```
which renders as
image

As to why they are different, I think @ev-br 's comment in gh-10328 [here](https://github.com/scipy/scipy/issues/10328#issuecomment-503485222) is relevant. 

Really, this issue seems to be a manifestation of the more general issue being discussed in gh-10328. It might be possible to define the one-sentence description, e.g. "Confidence interval with equal areas..." in one place and use it in both, but I'm not sure that it's worth the effort to implement just that. I would suggest that we just use this PR to make the `interval` method description in the subclass (e.g. `poisson`) documentation consistent with the description in `rv_continuous` for now, and focus on the more general issue in gh-10328.

**Review by mdhaber — APPROVED** [review]
I'd suggest merging as-is (maybe after adding a space in `[0, 1]`) as a temporary improvement, then focusing on gh-10328 (making sure there is complete subclass-specific documentation for all methods).

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def instancemethod(func, obj, cls):
 """
 _doc_interval = """\
 interval(alpha, %(shapes)s, loc=0, scale=1)
-    Endpoints of the range that contains alpha percent of the distribution
+    Endpoints of the range that contains fraction alpha [0,1] of the distribution
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:140]:
```suggestion
    Endpoints of the range that contains fraction alpha [0, 1] of the distribution
```

**Review by WarrenWeckesser — APPROVED** [review]
This simple doc change looks good.

**Comment by mdhaber** [comment]
Sorry all, not sure what happened here. I tried to merge, and something else happened instead. If the changes didn't take, I'll add them in a separate PR.

**Pull request closed without merging**